public/components/bulkReindexController.react.js

```diff
@@ -0,0 +1,208 @@
+            var json = JSON.parse(response);
+            var reindexInProgress =  json.IsReindexing;
+            if (!reindexInProgress) {
+                this.setState({
+                    inBulkMode: false
+                });
+            } else {
+                this.loadRunningReindexes(json);
+                this.setState({
+                    inBulkMode: true
+                });
+            }
+        });
+    }
+
+    cancelReindex(reindex, isReindexRunning) {
+        if (isReindexRunning) {
+            ContentSourceService.cancelReindex(reindex.id, reindex.env).then( response => {
+                    this.setState({
+                        runningReindexes: this.state.runningReindexes.filter(r => r !== reindex)
+                    });
+                },
+                errors => {
+                    this.setState({alertStyle: 'danger', alertMessage: 'Failed to cancel running reindex for '+ reindex.name, alertVisibility: true});
+                });
+        } else {
+            ContentSourceService.cancelPendingReindex(reindex.id, reindex.env).then( response => {
+                this.setState({
+                    pendingReindexes: this.state.pendingReindexes.filter(r => r !== reindex)
+                });
+            },
+                errors => {
+                    this.setState({alertStyle: 'danger', alertMessage: 'Failed to cancel pending reindex for ' + reindex.name, alertVisibility: true});
+                });
+        }
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        this.requestBulkStatus().then(response => {
+            var json = JSON.parse(response);
+            var reindexInProgress =  json.IsReindexing;
+            if (!reindexInProgress) {
+                let environments = this.generateEnvironments();
+                this.setState({alertStyle: 'success', alertMessage: 'Bulk job submitted.', alertVisibility: true});
+                this.initiateBulkReindex(environments);
+            } else {
+                this.setState({alertStyle: 'danger', alertMessage: 'Bulk job in progress. Please wait till job has finished.', alertVisibility: true});
+            }
+        });
+    }
+
+    handleReindexLiveStack(e) {
+        this.setState({reindexLiveStack: e.target.checked});
+    }
+
+    handleReindexPreviewStack(e) {
+        this.setState({reindexPreviewStack: e.target.checked});
+    }
+
+    handleReindexProd(e) {
+        this.setState({reindexProd: e.target.checked});
+    }
+
+    handleReindexCode(e) {
+        this.setState({reindexCode: e.target.checked});
+    }
+
+    render () {
+
+        return (
+            <div id="page-wrapper">
+                <div className="container-fluid">
+                    <div>
+                        { this.state.alertVisibility ? <Alert bsStyle={this.state.alertStyle}>{this.state.alertMessage}</Alert> : null }
+                        <form className="form-horizontal" onSubmit={this.handleSubmit}>
```

Comment:
@regiskuckaertz Should this not be `this.handleSubmit()` ? Get myself confused with javascript functions...